### PR TITLE
Don't generate scaffold for invalid models

### LINKED ISSUE #1482: Don't generate broken code for scaffolds
If you try to run the scaffold generator for a model with no editable fields the generated code will be broken (unused imports and syntax error in import statement)

Instead of generating broken code, a message could be printed about not finding any editable fields, and then we exit the generator before generating any code. Should probably also provide a `--force` option to run the generator anyway.

See #1435 for more context

### PR DESCRIPTION
**Pull request by alicelovescake**
### tl;dr

- [x] Resolves #1482 
- [x] Stops execution of scaffold generator if a model has no editable fields
- [x] Ensures `componentFiles` runs first so generator can exit early

Within the `scaffold.js` file, the `componentFiles` function is called first to ensure `fieldsToImport` is not empty. If it is, we throw an error message to let the user know that there are no editable fields in their model.

I didn't implement a force option because the generator will fail with an error anyways. Happy to provide further edits based on the team's feedback!

### CODE REVIEW

**Comment by Tobbe** [comment]
That's an old bug! Thanks for fixing it :)

**Comment by Tobbe** [comment]
@alicelovescake Can you please add a unit test for this as well?

**Comment by thedavidprice** [comment]
@alicelovescake I'm so very excited to see this PR! 🚀 Thank you for taking the time and diving into the Redwood waters with us. From here out we'll make sure a core team maintainer is assigned to review this and discuss with you step by step. We are about to release v0.34, which will take up our primary focus. But we'll make sure this is in the process for a future release. If we don't respond to you for more than a couple days, just nudge us here in the PR.

@Tobbe Alice joined one of the Contributing Workshops, so I know she's new to Redwood. If possible, could you point here in the direction of an existing unit test to use as a reference?

**Comment by alicelovescake** [comment]
@thedavidprice Thanks for remembering me from the workshop! I'm excited to contribute for the first time to Redwood. The welcoming community makes it a bit less intimidating. 😅

I appreciate the feedback to help me iterate on this PR. Know that you guys are busy.
@Tobbe Happy to add unit tests this week. Any examples would be great but I can also search through source code!

**Comment by Tobbe** [comment]
> @Tobbe Happy to add unit tests this week. Any examples would be great but I can also search through source code!

For the scaffold generator you'll be adding a test to `packages/cli/src/commands/generate/scaffold/__tests__/scaffold.test.js`. Next to the test files there's a `fixtures` folder with a `schema.prisma` file that the tests uses. I would create a model called `NoEditableFields` and use that in a test. The test could do something like `expect(...).toThrow('There are no editable fields in the NoEditableFields model')`

(You'll see a lot of the tests use `toMatchSnapshot()`. You won't use that for your test, but if you want to read more about that, you can do so here: https://jestjs.io/docs/snapshot-testing)

**Comment by alicelovescake** [comment]
@Tobbe Thanks for the clear instructions! I just added the test for the error being thrown with model of `NoEditableFields`. Sorry it took me a bit longer to get it done (exams). 

Let me know if you have any feedback for improvement! Appreciate your time :)

File: packages/cli/src/commands/generate/scaffold/__tests__/scaffold.test.js [hunk 1]
```diff
@@ -270,6 +270,20 @@ describe('in javascript (default) mode', () => {
...
+      async () =>
+        await scaffold.files({
+          ...getDefaultArgs(defaults),
+          model: 'NoEditableField',
+          tests: true,
+          nestScaffoldByModel: true,
+        })
+    ).rejects.toThrow(
+      `There are no editable fields in the NoEditableField model`
+    )
```
Comment (Tobbe) [inline packages/cli/src/commands/generate/scaffold/__tests__/scaffold.test.js:284]: I slightly updated the handling of promise based exception handling to match the `async/await` option in the docs https://jestjs.io/docs/expect#rejects (and also switched from \` to ')

```suggestion
    await expect(
      scaffold.files({
        ...getDefaultArgs(defaults),
        model: 'NoEditableField',
        tests: true,
        nestScaffoldByModel: true,
      })
    ).rejects.toThrow(
      'There are no editable fields in the NoEditableField model'
    )
```

**Comment by Tobbe** [comment]
@alicelovescake Thanks for your awesome work!

**Comment by alicelovescake** [comment]
@Tobbe Thanks for your guidance and feedback. You've made my first contribution experience with Redwood great! Excited to do more!

**Pull request merged**